packages/server/src/messages/api.ts

```diff
@@ -129,5 +142,65 @@ export class MessageApi extends ClientScopedApi {
         res.send({ count: deleted })
       })
     )
+
+    this.sockets.handle('messages.create', async (socket, message) => {
+      const { error } = CreateMsgSocketSchema.validate(message.data)
+      if (error) {
+        return this.sockets.reply(socket, message, { error: true, message: error.message })
+      }
+
+      const userId = this.socketService.getUserId(socket)
+      if (!userId) {
+        return this.sockets.reply(socket, message, {
+          error: true,
+          message: 'socket does not have user rights'
+        })
+      }
+
+      const { conversationId, payload } = message.data
+      const conversation = await this.conversations.get(conversationId)
+
+      if (!conversation) {
+        return this.sockets.reply(socket, message, { error: true, message: 'conversation does not exist' })
+      } else if (conversation.userId !== userId) {
+        return this.sockets.reply(socket, message, {
+          error: true,
+          message: 'conversation does not belong to that user'
+        })
+      }
+
+      const msg = await this.chat.send(conversationId, userId, payload, { socket })
+      this.sockets.reply(socket, message, msg)
```

Comment:
Not sure I understand this part. Both functions call socket.send!? What's the outcome?

Reply:
The from argument of the send function allows to avoid sending back the message to the connector that sent it (socket, channel, http client, whatever). So here we're passing the socket as an argument so send doesn't create a generic message for our socket. We want to respond to the socket request directly instead so we do it here with reply. This allows the frontend to know exactly that a message it received came from somewhere else or if it came from a request it made itself (and will have many advantages in the future, most notably it will allow use to display the sent message in the ui directly when it's written).